plugins/alarmclock/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/alarmclock/AlarmClockResource.java

```diff
@@ -73,16 +70,8 @@
 	@GET
 	@Path("alarms")
 	@Produces(MediaType.APPLICATION_JSON)
-	public Timestamp[] getAllAlarms() {
-		List<Alarm> alarms = this.logic.getAllAlarms();
-		Timestamp[] timestamps = new Timestamp[alarms.size()];
-		for (int i = 0; i < alarms.size(); i++) {
-			if (alarms.get(i) != null) {
-				timestamps[i] = new Timestamp(alarms.get(i));
-				timestamps[i].setLink(createAlarmPath(alarms.get(i).getId()));
-			}
-		}
-		return timestamps;
+	public List<Alarm> getAllAlarms() {
+		return this.logic.getAllAlarms();
```

Comment:
each alarm should have a link set that a client can use to access each alarm individually